### PR TITLE
Support customizing the SDS socket filename used by the agent

### LINKED ISSUE #48845: Configure SDS socket filename through configuration
**Describe the feature request**

 Envoy SDS defined socket path is  "/run/secrets/workload-spiffe-uds/socket",  we need a way to configure it since sometimes we may encounter mounting issues, example in distroless 

**Describe alternatives you've considered**

**Affected product area (please put an X in all that apply)**

[] Ambient
[ ] Docs
[ ] Dual Stack
[ ] Installation
[ ] Networking
[ ] Performance and Scalability
[ ] Extensions and Telemetry
[x] Security
[ ] Test and Release
[ ] User Experience
[ ] Developer Infrastructure

**Affected features (please put an X in all that apply)**

[ ] Multi Cluster
[ ] Virtual Machine
[ ] Multi Control Plane

**Additional context**

### PR DESCRIPTION
**Pull request by bleggett**
Fixes https://github.com/istio/istio/issues/48845 and removes the currently-documented requirement to force the SPIRE SDS server use our Istio-default SDS socket name, versus whatever the (user-configurable) SPIRE SDS server socket filename happens to be.

Note that this is still an issue even if you use the SPIRE CSI driver like we strongly suggest you do - the CSI driver will handle translating the path for you, but not the socket filename.

This introduces `WORKLOAD_IDENTITY_SOCKET_FILE` as an agent env var.

- If this is **set** to a non-default value, the agent will expect to find a non-Istio SDS server socket at the _hardcoded_ path:

`WorkloadIdentityPath/WORKLOAD_IDENTITY_SOCKET_FILE`

and will throw an error if no healthy socket found. Otherwise it will listen to it.

- If this is **unset**, the agent will do what it did before, which is start an Istio default SDS server instance with a hardcoded path and hardcoded socket file of:

`WorkloadIdentityPath/DefaultWorkloadIdentitySocketFile`

and listen to it.

As a side effect, this **removes/replaces** the agent env var `USE_EXTERNAL_WORKLOAD_SDS` (which was added in https://github.com/istio/istio/pull/45941) - the same behavior can be obtained by explicitly setting `WORKLOAD_IDENTITY_SOCKET_FILE` to a non-default value.

---

Q: "Why not make the whole path configurable via env var?"

A: One, that would make hijacking workload identity by overriding env vars to any path much easier. This way you can't hijack the path, just the filename - which is relatively trivial to defend against. Two, we want people to use the CSI driver, or K8S volume mounts, to map their host sockets to the (fixed istio location), as a matter of simplicity.

### CODE REVIEW

File: pkg/istio-agent/agent.go [hunk 1]
```diff
@@ -351,20 +355,47 @@ func (a *Agent) Run(ctx context.Context) (func(), error) {
...
+	// is found AND no custom SDS socket path is configured.
+	//
+	//
+	// We do that like so
+	// 1. compare the (hardcoded) Istio default SDS server path with the (configurable) client path.
+	// 2. if they are different, do not start the Istio default SDS server.
+	// 3. if they are different, and no socket exists at that location, error out.
+	// 4. if they are not different, start the Istio default SDS server and use it.
+
+	// Correctness check - we do not want people sneaking paths into this
```
Comment (howardjohn) [inline pkg/istio-agent/agent.go:372]: I'm not sold on this. There isn't really a security violation - pod owners have total control over their pod. And I don't want to have to have yet another tweak to this code path in 6 months when someone needs a custom path because we were too opinionated. WDYT?

**Review by howardjohn — COMMENTED** [review]
Looks good !

**Comment by m3r00t** [comment]
Team - can I understand why am I receiving all the threads ?

I believe the request is to auto fwd to my email ?

Best
Love

File: pkg/istio-agent/agent.go (hunk 1 above)
Comment (bleggett) [reply pkg/istio-agent/agent.go:372]: I swear there was a conversation about this where this was brought up as a sticking point for doing this previously, but I can't remember/find it.

> And I don't want to have to have yet another tweak to this code path in 6 months when someone needs a custom path because we were too opinionated. WDYT?

That occurred to me. I am slightly in favor of having it be fixed/opinionated here tho, for the following reasons.

- All the other paths in `security.go` are already similarly hardcoded/nonconfigurable, including the SDS ones. It's your pod yes, but our mount, and our sidecar.
- We document and support and recommend doing this in exactly one way - via the SPIRE CSI driver. And if you use that, you by-definition never need to alter this path - making the whole path configurable just makes it easier to not-follow our docs and do something that is less secure and operationally harmful that we do not support or recommend (raw hostmounts, no CSI driver). I literally want to make it hard to do something that both SPIRE and us explicitly recommend against and (in our case) do not currently document or support.


basically, there is no valid case for using an alt path other than to avoid the SPIRE CSI driver, which is a deployment model we explicitly do not recommend or document.

Comment (howardjohn) [reply pkg/istio-agent/agent.go:372]: alright fine with me, I'm convinced. One nice thing is we can keep the same API and allow full path later if needed

**Review by howardjohn — APPROVED** [review]
